### ISSUE TITLE
Using on Android

### ISSUE THREAD

**Issue by SuppSandroB** [StevenBlack/hosts#134]
Just let you know that can be used on Android in Drony app.
https://www.youtube.com/watch?v=kyc0Nw58rkE

**Comment by StevenBlack** [comment]
Well how cool is that!   Thanks for sharing @SuppSandroB !

I'm going to link this in the readme files....

**Comment by SuppSandroB** [comment]
Will also implement automatic fetching and refreshing local list in next versions.

**Comment by soredake** [comment]
I user this script 
```
cd $HOME/git/hosts && peth=$(realpath .)  && git reset --hard && git pull && chmod +x updateHostsFile.py && python3 updateHostsFile.py --extensions gambling -a && /system/bin/su -c "mount -o remount,rw /system" && /system/bin/su -c "mv $peth/hosts /etc/hosts" && /system/bin/su -c "mount -o remount,ro /system"
```

**Comment by mzilikazi** [comment]
@soredake
Sweet! It dawned on me today that this was do-able and here it is!

**Comment by 07416** [comment]
@SuppSandroB Is Drony only for non-rooted users?

**Comment by konarkmodi** [comment]
I was facing similar issue, so I did a trick which I would love to receive a feedback on. I am using MITMProxy and put that in the proxy setting on my Android, the hosts file that get's resolved is the one on the MITMProxy machine.

**Comment by zamber** [comment]
@07416, Drony will work for non-rooted devices but you have to keep the local VPN service active for it to work. NoRoot Firewall works on the same premise by giving you information about what apps use what data. Sadly, you can't have both active at the same time.

**Comment by matijagrcic** [comment]
Setup [Android Debug Bridge (ADB) and USB debugging](https://developer.android.com/studio/command-line/adb.html)

```
adb devices -l
adb pull /system/etc/hosts C:\hosts
```

Modify then push the hosts back to your android device
`adb push C:\hosts /system/etc/`

**NOTE : Your device needs to be rooted.**

**Comment by kronflux** [comment]
I have emailed @StevenBlack a script to create a flashable zip for under android(TWRP in particular, although it should work with other recoveries), it will be up to him whether or not he wishes to include it in this project, but at the very least I'm hoping some inspiration comes from it :)

**Comment by gfyoung** [comment]
I think <a href="https://github.com/AdAway/AdAway">AdAway/AdAway</a> does the job already?

**Comment by Jezza-x3** [comment]
@kronflux What would the zip even do?

**Comment by gfyoung** [comment]
@Jezza-x3 : The idea is that you can install the ZIP from recovery mode on Android.

**Comment by Jezza-x3** [comment]
@gfyoung I know what the process is and all. I just don't understand what @kronflux wants the zip to do.

**Comment by gfyoung** [comment]
@Jezza-x3 : Misinterpreted your question there.  I think he wants to install a new hosts file IIUC.

**Comment by kronflux** [comment]
@gfyoung @Jezza-x3 , the idea is to replace the actual linux hosts file within an android device, just as you would on a windows or linux system.
For the record, I've already done this, however it was rejected for official support by @StevenBlack due to his lack of an android device for testing.
You can find it here:
https://github.com/kronflux/hosts-android

Basically it's a script that creates an android recovery flashable zip(from TWRP or similar recovery) which replaces the hosts file, using the hosts file in the root directory of this repo.

**Comment by gfyoung** [comment]
Okay, that's what I thought.  Unless there is an easy way to automate such tests, @StevenBlack I think this issue can be closed (with pointing to your repository @kronflux ?)

**Comment by kronflux** [comment]
Yep. I concur, and I'll end on the note that it works flawlessly for all my android devices! I hope someone else finds it useful ^_^

**Comment by Jezza-x3** [comment]
What is the reason of this instead of using a file explorer to replace it?

**Comment by gfyoung** [comment]
Because automation is beautiful? 😄

**Comment by kronflux** [comment]
Avoids the need for root access, as well as being a direct way to take affect immediately, rather than having to reboot after modifying. This way, you only boot once after flashing.
And yes, automation is beautiful. :D

**Comment by Jezza-x3** [comment]
Still, needs a custom recovery menu though correct?

**Comment by vdias** [comment]
I'm using this list in dns66 app... I think it is the best Android solution.

**Comment by Jezza-x3** [comment]
@vdias can you link it? I can't find it

**Comment by Mahaveer-Jeevagan** [comment]
https://github.com/julian-klode/dns66/blob/master/README.md

**Comment by Jezza-x3** [comment]
I've seen other apps like this and all but the problem for me is I can't use my VPN as this creates one and would stop the one I'm using.

**Comment by StevenBlack** [comment]
@kronflux I'm considering dropping `.zip` file generation and propagation in this repo.

See [this comment](https://github.com/StevenBlack/hosts/issues/272#issuecomment-322204496) in issue #272.

Does this affect you?  I presume you're **not** using the `hosts,.zip` file from this repo, just `hosts`.  Is that correct?

Just checking 🔎

**Comment by Bokoblin** [comment]
On Android, disqus internal links from emails or from web fails to be loaded because "https://redirect.viglink.com/" seems to be blocked by this module

**Comment by StevenBlack** [comment]
@Bokoblin Arthur, `redirect.viglink.com` is flagged by both mvps.org and someonewhocares.org which are longstanding, vigorously curated sources.

Sites such as Disquis can be popular and scummy at the same time.

You may want to take this up with [mvps.org](https://github.com/StevenBlack/hosts/blob/master/data/mvps.org/update.json) and [someonewhocares.org](https://github.com/StevenBlack/hosts/blob/master/data/someonewhocares.org/update.json).  I trust those guys to do the right thing, and I'm inclined to believe `redirect.viglink.com` is blocked for solid reasons.
